### PR TITLE
raft topology: fixes after #13884

### PR DESCRIPTION
**Pull request by gusev-p**
This PR fixes some problems found after the PR was merged:
  * missed `node_to_work_on` assignment in `handle_topology_transition`;
  * change error reporting in `update_fence_version` from `on_internal_error` to regular exceptions, since that exceptions can happen during normal operation.
  * `update_fence_version` has beed moved  after `group0_service.setup_group0_if_exist` in `main.cc`, otherwise we use uninitialized `token_metadata::version` and get an error.

Fixes: #14303

### CODE REVIEW

**Review by tgrabiec — APPROVED** [review]

File: locator/token_metadata.cc [hunk 1]
```diff
@@ -1192,13 +1192,14 @@ void shared_token_metadata::set(mutable_token_metadata_ptr tmptr) noexcept {
 
 void shared_token_metadata::update_fence_version(token_metadata::version_t version) {
     if (const auto current_version = _shared->get_version(); version > current_version) {
-        on_internal_error(tlogger,
+        throw std::runtime_error(
```
[the remark this answers is not in the document]
Comment (gusev-p) [reply locator/token_metadata.cc:1195]: True

[the remark this answers is not in the document]
Comment (kbr-scylla) [reply locator/token_metadata.cc:1195]: So should we change it back to `on_internal_error`?

[the remark this answers is not in the document]
Comment (gusev-p) [reply locator/token_metadata.cc]: yes, updated

**Review by kbr-scylla — APPROVED** [review]
Thanks.
Also thanks for the nice comments - now it's perfect.

**Pull request merged**